hadoop-ozone/client/src/test/java/org/apache/hadoop/ozone/client/TestReplicatedFileChecksumHelper.java

```diff
@@ -0,0 +1,223 @@
+    assertTrue(fileChecksum instanceof MD5MD5CRC32GzipFileChecksum);
+    assertEquals(DataChecksum.Type.CRC32,
+        ((MD5MD5CRC32GzipFileChecksum)fileChecksum).getCrcType());
+  }
+
+  @Test
+  public void testOneBlock() throws IOException {
+    // test the file checksum of a file with one block.
+    OzoneConfiguration conf = new OzoneConfiguration();
+
+    RpcClient rpcClient = Mockito.mock(RpcClient.class);
+
+    List<DatanodeDetails> dns = Arrays.asList(
+        DatanodeDetails.newBuilder().setUuid(UUID.randomUUID()).build());
+    Pipeline pipeline;
+    pipeline = Pipeline.newBuilder()
+        .setId(PipelineID.randomId())
+        .setReplicationConfig(
+            new RatisReplicationConfig(HddsProtos.ReplicationFactor.THREE))
+        .setState(Pipeline.PipelineState.CLOSED)
+        .setNodes(dns)
+        .build();
+
+    XceiverClientGrpc client = new XceiverClientGrpc(pipeline, conf) {
+      @Override
+      public XceiverClientReply sendCommandAsync(
+          ContainerProtos.ContainerCommandRequestProto request,
+          DatanodeDetails dn) {
+        return buildValidResponse();
+      }
+    };
+    XceiverClientFactory factory = Mockito.mock(XceiverClientFactory.class);
+    when(factory.acquireClientForReadData(any())).thenReturn(client);
+
+    when(rpcClient.getXceiverClientManager()).thenReturn(factory);
+
+    OzoneManagerProtocol om = Mockito.mock(OzoneManagerProtocol.class);
+    when(rpcClient.getOzoneManagerClient()).thenReturn(om);
+
+    BlockID blockID = new BlockID(1, 1);
+    OmKeyLocationInfo omKeyLocationInfo =
+        new OmKeyLocationInfo.Builder().setPipeline(pipeline)
+            .setBlockID(blockID)
+            .build();
+
+    List<OmKeyLocationInfo> omKeyLocationInfoList =
+        Arrays.asList(omKeyLocationInfo);
+
+    OmKeyInfo omKeyInfo = new OmKeyInfo.Builder()
+        .setVolumeName(null)
+        .setBucketName(null)
+        .setKeyName(null)
+        .setOmKeyLocationInfos(Collections.singletonList(
+            new OmKeyLocationInfoGroup(0, omKeyLocationInfoList)))
+        .setCreationTime(Time.now())
+        .setModificationTime(Time.now())
+        .setDataSize(0)
+        .setReplicationConfig(new RatisReplicationConfig(
+            HddsProtos.ReplicationFactor.ONE))
+        .setFileEncryptionInfo(null)
+        .setAcls(null)
+        .build();
+
+    when(om.lookupKey(any())).thenReturn(omKeyInfo);
+
+    OzoneVolume volume = Mockito.mock(OzoneVolume.class);
+    when(volume.getName()).thenReturn("vol1");
+    OzoneBucket bucket = Mockito.mock(OzoneBucket.class);
+    when(bucket.getName()).thenReturn("bucket1");
+
+    ReplicatedFileChecksumHelper helper = new ReplicatedFileChecksumHelper(
+        volume, bucket, "dummy", 10, rpcClient);
+
+    helper.compute();
+    FileChecksum fileChecksum = helper.getFileChecksum();
+    assertTrue(fileChecksum instanceof MD5MD5CRC32GzipFileChecksum);
+    assertEquals(1, helper.getKeyLocationInfos().size());
```

Comment:
Please use `helper.getKeyLocationInfoList()` and we can remove additional get method `BaseFileChecksumHelper#getKeyLocationInfos()`,  right?


```
BaseFileChecksumHelper.java 
  @VisibleForTesting
  List<OmKeyLocationInfo> getKeyLocationInfos() {
    return keyLocationInfos;
  }
```